### ISSUE TITLE
Lexica Botania not unlocking chapters/re-locking after finally unlocking

### ISSUE THREAD

**Issue by Mcswaggermama** [Vazkii/Botania#3631]
# Version Information
Forge version:  0.162.0.12 (64 bit)
Botania version:  411th edition

# Further Information
no crash

What I expected to happen:
 to progress through the game finishing advancements and unlocking **generating flora** along with other chapters

What happened instead: 
For a while the book bugged out and didn't count me making a petal apothecary or pure daisies[marking the tabs with question marks], suddenly after a corruption and backup load of a previous version of the world i continued and unlocked the **generating flora**[somehow, i still dont know what i did] chapter and the bugs in the basics chapter went away, after stepping away from the world and loading it up a few hours later the bug returned and the **generating flora** chapter locked once again and the bugged question marks over petal apothecary and pure daisy returned along with one over the runic altar. I have made and used all of these items along with open crates, mana spreaders, mana pools, endoflames, the lava eating flower, the sound reducing flower, the agricultural speed flower, trinkets, mana tablets, and items to craft runes such as mana power, mana pearls, mana steel, mana diamonds, and i have made advancements past another world where everything worked fine. **Now**, this _is_ in a modpack (Enigmatica 6 1.16.5) however i have had the same issue with another modpack[save for the re-locking of an unlocked chapter] (all the mods 6) so i feel like it is an issue with the mod itself? I'm not sure, and i can take this to those mod fourms if the team of botania wishes. The only difference is that in the world where i experienced no bugs was in All the Mods 6 in my _own_ world, whereas the two worlds with glitches have been happening in a lan server not originated to my computer.

I may just be missing something and not seeing it and the mod is working as intended, but that just leaves the fact of this is all really confusing and frustrating and i really just want the mod and the book to work so i can read about the generating flora without ruining the experience by googling it online. I haven't found anything really about what specific actions unlock what chapters and logs and i think that if the mod is working as intended that might be a nice touch? Because the unlock bar at the bottom left of every chapter only says how much you've unlocked and not how to unlock it, and the advancements dont really help because i've tried doing those but as i've said i have made more advancements in the super bugged current world i'm on than the one where i've had no issues.

TLDR; i cant freaking get generating flora to unlock because i cant get the question marks over some stuff in the basics chapter to go away, then suddenly i did[no idea what i did], and then suddenly it re-locked >:O

**Comment by Hubry** [comment]
Locks in the lexicon are done through advancements, some of which are hidden. Tell me what the following commands say (of course don't forget the leading slash):
https://github.com/Vazkii/Botania/blob/294023d501d2958c32dfebd918595da3434073f2/src/main/resources/data/botania/functions/full_book.mcfunction#L1-L4

The question marks are not all handled by these, the generating flower entries specifically should be looking at `botania:main/generating_flower`. Test granting that one too.

**Comment by NeosGhost** [comment]
Having a similar problem as well
Forge: 35.1.36
Botania: 1.16.4-411

runic altar and portal to alfheim have a orange marker next to them. unlocked elder knowledge from the portal just cant get those to go. 
![image](https://user-images.githubusercontent.com/16095247/106702782-90069f00-65ae-11eb-960a-c67ea0dcd690.png)

**Comment by Putsrox** [comment]
Hey @Hubry !

Forge: v. 0.167.2.4
Botania-1.16.4-411

Have the same issue on runic altar and pure daisy.
Tried the commands you said above and even the one that grants the full book. Still having the issue. 

Here is the result of the commands you requested
![bug botania](https://user-images.githubusercontent.com/78773924/107309818-1590c000-6a83-11eb-9714-853afde7641e.PNG)

Hope this helps

**Comment by Hubry** [comment]
Alright, what modpack and can you provide a world? (you can strip most chunks)

**Comment by Putsrox** [comment]
Hi again!

Modpack: All the Mods 6 - ATM6 - 1.16.5

unf, cant provide world :/

**Comment by Hubry** [comment]
Oh well, I'll try to reproduce soon without that - hopefully it doesn't have weird requirements like the lexicon trading bug had.

**Comment by Putsrox** [comment]
Hi again!

So got some more information for you, @Hubry.
I'm playing with my girlfriend. She's hosting and I go to server via LAN. We tried three different worlds and happens on all of them but only to me. She doesn't get this bug.
Also, it happens when I log out of the server and come back in.
In other words I can do whatever I want and reach as deep in the mod as I want and it will be ok. However, when I log out (even if the server stays up) and come back online I'll have the bug. It happens on 3 different "achievements" Runic altar, pure daisy and petal apothecary.

Hope this helps

**Comment by Hubry** [comment]
Seems like it happens only to hidden advancements, some mod might be trying to optimize stuff by not sending them, kinda similiar to what was experienced with Patchouli and Triumph on 1.12...

For now, just disable advancement locking for `botania:lexicon` in Patchouli's config.

**Comment by Putsrox** [comment]
Hi again,

Small update. 
Created a new world and I'm the host.
The bug doesn't happen again.

Therefore, might be something with logging into LAN worlds?

**Comment by ArthurBlink** [comment]
Hi, I found the answer...

Try to revoke your bugged advancement and later got it again.

For example:

- First type this command this: 

/advancement revoke @s only botania:main/runic_altar_pickup

(This for the Runic Altar Case) 

Later:  Break your Runic Altar and put ir again in floor.

<3

**Comment by Kandrathe** [comment]
It wasn't apparent, but making the cacophonium unlocked my progression to Terrasteel.

**Comment by ArthurBlink** [comment]
> It wasn't apparent, but making the cacophonium unlocked my progression to Terrasteel.

In the moment i've tried to craft all possible things (cacophonium included) but nothing worked, I keep thinking that is a advancements bug, but only revoking it and gotting it again like in my last answer, and finally I could fix it.

**Comment by romangraef** [comment]
I feel like this is an issue with patchouli in Open to LAN environments. I have the same issue an a custom modpack that is Open to LAN (modlist is basically just botania, JEI, and refined storage. Botania version 414, 1.16.5)

**Comment by Richard-Rogalski** [comment]
Friend has been having this issue. LAN world, and it's been happening to him constantly. I'm the host. He's constantly cheating in advancements that he should've gotten, even multiple times for the same one. 
Botania 1.16.5-416
Forge 36.1.12

**Comment by Hubry** [comment]
Looks like the common thing between all occurences here is LAN worlds.

**Comment by Jacesrhapsody** [comment]
I'm having the same issue with the "Portal to Alfheim" hidden advancement. The fact that it's a hidden advancement I think makes the above workaround from ArthurBlink not work, since the system doesn't seem to recognize the "elf_lexicon_pickup" tag. I also tried the function to give all the advancements and that also didn't work.

I'm playing with the Valhelsia 3 modpack on a Bisect server with a couple friends, so it still might be related to the above issue (looked like to pinpoint on LAN worlds), but it could also be a number of other things due to the size of the modpack. I'm unfortunately super illiterate when it comes to this sort of thing, so I don't know how to disable advancement locking in Patchouli's config like Hubry suggested above.

Anyways, any advice would be appreciated.

**Comment by Jacesrhapsody** [comment]
Well, I logged in today and the problem seems to have somehow resolved itself.

**Comment by silencethrills** [comment]
Me and my friends recently started having this problem as well, and it does involve being apart of a LAN world. 
When ever we start it up, the person's book restarts their progress throughout the book and it proceeds to do this every time they join the server or disconnect and come back.

**Comment by nonchip** [comment]
just happened to my girlfriend (playing in my lan world; modpack: atm7sky), where it re-locked the "petal apothecary" and "runic altar" ones after she already made them (and got them unlocked).

**Comment by Eugene21040** [comment]
1,19,2 The problem remains, the solution to the problem of obtaining an achievement

But after restarting the world, the problem returns

**Comment by TheRealWormbo** [comment]
Since this is very likely an issue with Patchouli, not in Botania, I'm going to close this bug.
